streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -132,18 +131,18 @@ private void determineIsQueryable(final String queryableStoreName) {
         ChangedDeserializer<? extends V> changedValueDeserializer = new ChangedDeserializer<>(valueDeserializer);
 
         // send the aggregate key-value pairs to the intermediate topic for partitioning
-        topology.addInternalTopic(topic);
-        topology.addSink(sinkName, topic, keySerializer, changedValueSerializer, this.name);
+        builder.internalTopologyBuilder.addInternalTopic(topic);
```

Comment:
Okay this will be the place that I mentioned about the place that `Topology's public API is not sufficient`. One way we can walk around it is to introduce one constructor in `Topology` which takes a `InternalTopologyBuilder` and the current constructor will need to be explicit and create the 

```
final InternalTopologyBuilder internalTopologyBuilder = new InternalTopologyBuilder();
```

The added constructor will also be package-private so it is not exposed to users, but `o.a.k.streams. StreamsBuilder` can use it, so it can create the `internalTopologyBuilder` and pass it to the `Topology` in its own constructor, so that it can then hold on its reference to use it in places like here.

BTW as I mentioned above, for classes we do not really need to access the `internalTopologyBuilder` we should restrict ourselves to only call `Topology`'s public APIs.